src/game/AuctionHouseBot/AuctionHouseBot.cpp

```diff
@@ -1809,11 +1809,18 @@ void AuctionBotSeller::SetPricesOfItem(AHB_Seller_Config& config, uint32& buyp,
     double temp_buyp = buyp * stackcnt *
                        (itemQuality < MAX_AUCTION_QUALITY ? config.GetPriceRatioPerQuality(AuctionQuality(itemQuality)) : 1) ;
 
-    double randrange = temp_buyp * 0.4;
-    buyp = (urand(temp_buyp - randrange, temp_buyp + randrange) / 100) + 1;
+    double randrange = temp_buyp * 0.4;    
+    uint32 buypMin = temp_buyp - randrange;
+    uint32 buypMax = temp_buyp + randrange;
+    InvertMinMaxIfNeeded(buypMin, buypMax);
+    buyp = (urand(buypMin, buypMax) / 100) + 1;
+    
     double urandrange = buyp * 40;
     double temp_bidp = buyp * 50;
-    bidp = (urand(temp_bidp - urandrange, temp_bidp + urandrange) / 100) + 1;
+    uint32 bidPmin = temp_bidp - urandrange;
+    uint32 bidPmax = temp_bidp + urandrange;
+    InvertMinMaxIfNeeded(bidPmin, bidPmax);
+    bidp = (urand(bidPmin, bidPmax) / 100) + 1;
```

Comment:
This seems like a lot of variables to define.  My understanding of the issue is that sometimes randrange is negative and results in bad values being fed into `urand`.  If that's the case could the bug be mitigating by using the absolute value of `randrange`?

Reply:
perhaps an explicit casting for the urand method and a modulus on the uint32 max for the other issues you mentioned in discord.

Reply:
Perhaps make a check and cap value to max if overflow is detetected

Reply:
@leprasmurf @billy1arm 
The PR is cleaned-up